truss/templates/shared/middleware.py

```diff
@@ -0,0 +1,47 @@
+from typing import Optional, Tuple
+
+from msgpack_asgi import MessagePackMiddleware
+from shared.serialization import truss_msgpack_deserialize, truss_msgpack_serialize
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.requests import Request
+from starlette.types import ASGIApp
+
+
+class BinaryHeaderMiddleware(BaseHTTPMiddleware):
```

Comment:
Worth noting the following when using `BaseHTTPMiddleware` 

>Using BaseHTTPMiddleware will prevent changes to [contextlib.ContextVar](https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar)s from propagating upwards. That is, if you set a value for a ContextVar in your endpoint and try to read it from a middleware you will find that the value is not the same value you set in your endpoint (see [this test](https://github.com/encode/starlette/blob/621abc747a6604825190b93467918a0ec6456a24/tests/middleware/test_base.py#L192-L223) for an example of this behavior).

We don't do this right now but if we did, using this middleware might mutate / remove these values.